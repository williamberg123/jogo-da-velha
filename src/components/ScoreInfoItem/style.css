.Bar-group {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 80px;
    padding: 5px 10px;
    border-radius: 10px;
}

p {
    visibility: hidden;
    font-size: 0.7rem;
    margin-bottom: 10px;
    font-weight: bold;
}

.Bar-group img {
    width: 90%;
}

.Bar-score {
    margin-top: 10px;
    font-weight: bold;
    color: rgb(240, 84, 22);
    font-size: 1.2rem;
}

.whoArePlaying {
    background-color: rgba(85, 85, 85, 0.4);
}

.whoArePlaying p {
    visibility: visible;
}