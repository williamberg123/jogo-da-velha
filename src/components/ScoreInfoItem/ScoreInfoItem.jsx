import React from 'react';

import PropTypes from 'prop-types';

import Image from '../Image/Image';

import './style.css';

export default function ScoreInfoItem({ whoArePlaying, score, imageRoot }){
    return (
        <div className={`Bar-group ${!!whoArePlaying && (whoArePlaying)}`}>
            <p>Sua vez</p>
            <Image imageRoot={imageRoot} />
            <div className="Bar-score">{score}</div>
        </div>
    );
}

ScoreInfoItem.propTypes = {
    whoArePlaying: PropTypes.string.isRequired,
    score: PropTypes.number.isRequired,
    imageRoot: PropTypes.string.isRequired
};
