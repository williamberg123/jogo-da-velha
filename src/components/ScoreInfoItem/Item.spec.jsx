import React from 'react';
import { screen, render } from '@testing-library/react';
import ScoreInfoItem from './ScoreInfoItem';

describe('<ScoreInfoItem />', () => {
    it('should render element correctly', () => {
        const {debug} = render(<ScoreInfoItem whoArePlaying="cross" score={10} imageRoot="img/img.jpg" />);
        // debug();

        expect(screen.getByText('Sua vez')).toBeInTheDocument();
        expect(screen.getByRole('img', { name: 'img' })).toBeInTheDocument();
        expect(screen.getByText(/10/)).toBeInTheDocument();
    });

    it('should to have class cross when whoArePlaying is cross', () => {
        render(<ScoreInfoItem whoArePlaying="cross" score={5} imageRoot="img/img.jpg" />);

        expect(screen.getByText('Sua vez').parentElement).toHaveClass('cross');
    });

    it('should match snapshot', () => {
        render(<ScoreInfoItem whoArePlaying="cross" score={10} imageRoot="img/img.jpg" />);

        expect(screen.getByText('Sua vez').parentElement).toMatchSnapshot();
    });
});
