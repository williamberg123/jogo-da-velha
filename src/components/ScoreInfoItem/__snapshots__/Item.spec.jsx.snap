// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScoreInfoItem /> should match snapshot 1`] = `
<div
  class="Bar-group cross"
>
  <p>
    Sua vez
  </p>
  <img
    alt="img"
    src="img/img.jpg"
  />
  <div
    class="Bar-score"
  >
    10
  </div>
</div>
`;
